doc/services/llext/config.rst

```diff
@@ -0,0 +1,129 @@
+Configuration
+#############
+
+The following Kconfig options are available for the LLEXT subsystem:
+
+.. _llext_kconf_heap:
+
+Heap size
+----------
+
+The LLEXT subsystem needs a static heap to be allocated for extension related
+data. The following option controls this allocation.
+
+:kconfig:option:`CONFIG_LLEXT_HEAP_SIZE`
+
+        Size of the LLEXT heap in kilobytes.
+
+.. note::
+
+   When :ref:`user mode <usermode_api>` is enabled, the heap size must be
+   large enough to allow the extension sections to be allocated with the
+   alignment required by the architecture.
+
+.. _llext_kconf_type:
+
+ELF object type
+---------------
+
+The LLEXT subsystem supports loading different types of extensions; the type
+can be set by choosing among the following Kconfig options:
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_OBJECT`
+
+        Build and expect relocatable files as binary object type for the LLEXT
+        subsystem. A single compiler invocation is used to generate the object
+        file.
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_RELOCATABLE`
+
+        Build and expect relocatable (partially linked) files as the binary
+        object type for the LLEXT subsystem. These object files are generated
+        by the linker by combining multiple object files into a single one.
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_SHAREDLIB`
+
+        Build and expect shared libraries as binary object type for the LLEXT
+        subsystem. The standard linking process is used to generate the shared
+        library from multiple object files.
+
+        This is not currently supported on ARM architectures.
+
+.. _llext_kconf_storage:
+
+Minimize allocations
+--------------------
+
+The LLEXT subsystem loading mechanism, by default, uses a seek/read abstraction
+and copies all data into allocated memory; this is done to allow the extension
+to be loaded from any storage medium. Sometimes, however, data is already in a
+buffer in RAM and copying it is not necessary. The following option allows the
+LLEXT subsystem to optimize memory footprint in this case.
+
+:kconfig:option:`CONFIG_LLEXT_STORAGE_WRITABLE`
+
+        Allow the extension to be loaded by directly referencing section data
+        into the ELF buffer. To be effective, this requires the use of an ELF
+        loader that supports the ``peek`` functionality, such as the
+        :c:struct:`llext_buf_loader`.
+
+        .. warning::
+
+           The application must ensure that the buffer used to load the
+           extension remains allocated until the extension is unloaded.
+
+        .. note::
+
+           This will directly modify the contents of the buffer during the link
+           phase. Once the extension is unloaded, the buffer must be reloaded
+           before it can be used again in a call to :c:func:`llext_load`.
+
+        .. note::
+
+           This is currently *required* by the Xtensa architecture.
```

Comment:
If this is true we should file a bug/feature issue imo if we haven't already for this and maybe reference it here